Test/Mono.Cecil.Tests/ModuleTests.cs

```diff
@@ -294,5 +294,28 @@ public void ReadAndWriteFile ()
 			using (var module = ModuleDefinition.ReadModule (path))
 				Assert.AreEqual ("Foo.Foo", module.Types [1].FullName);
 		}
+
+		[Test]
+		public void ExceptionInWriteDoesNotKeepLockOnFile ()
+		{
+			var path = Path.GetTempFileName ();
+
+			var original = ModuleDefinition.CreateModule ("FooFoo", ModuleKind.Dll);
```

Comment:
The module `original` here is freshly created and doesn't hold to a backing `Stream`.
Calling `Write(string)` means Cecil creates and owns the stream to write to. Cecil crashes, and leaves the stream open while it failed to write to it.

I think the issue is valid.